### ISSUE TITLE
User Permissions do not remain when the page refreshes

### ISSUE THREAD

**Issue by aimee-mcneil-melville** [msupply-foundation/open-msupply#5860]
## What went wrong? 😲

Whilst adding permissions to scanning for #5223 I noticed that when the page refreshes/I navigated away and back, what had previously been blocked by the permissions checks was now allowed.

This was noticed in ```Cold Chain -> Equipment``` with the ```New Asset``` button, and in Detail View on the ```Update Status``` button

Re-synching reinstated the permissions and prevented accessing the button, until the next refresh.

**This is intermittent!**
I could not replicate this bug the next day, however @noel-yeldos could, and from a different branch. I am unsure if other areas of the app could also be affected - it's possibly to do with caching of the ```permissions``` from graphql?

This occurred in browser, I did not come across it whilst testing on tablet. 

## Expected behaviour 🤔
Permissions to remain consistent regardless of navigation or page refreshes.

Eg: if a user is prevented from adding an asset, they should remain prevented from adding an asset after refreshing the page

## How to Reproduce 🔨

Steps to reproduce the behaviour:

- [ ] Set user permissions in 4D: have user permission on for View Asset and off for Add/Edit Asset
- [ ] Sync ->  Go to Cold Chain -> Equipment
- [ ] Click ```New Asset```  - you should see a notification "You do not have permission to add an asset"
- [ ] Refresh the page and try New Asset again
- [ ] If you see the modal open for adding an asset, this is the bug!

## Your environment 🌱

- Open mSupply Version: 2.5.0
- Legacy mSupply Central Server Version: 

- Platform: Google Chrome (browser for Mac)

- Database type:

**Comment by andreievg** [comment]
Need a consistent way to replicate really, maybe try with network throttling ?

**Comment by lache-melvin** [comment]
Assigning to Korimako - timebox an hour to try to replicate issue, if we have consistent replicate can discuss severity of issue + solution 🙏

**Comment by noel-yeldos** [comment]
@lache-melvin Was able to replicate it 
https://github.com/user-attachments/assets/1a36a0dc-baee-4695-b253-714eab8fbd3c

**Comment by lache-melvin** [comment]
@noel-yeldos I'm not getting the same thing 🤔 Is this... only the first time you remove the permission? Different with throttling or something?

**Comment by lache-melvin** [comment]
If it's happening reliably for you, can you track down where in the code is not returning the right permission validation? Assuming its the `userHasPermission` in auth context... do you have a stale cookie that still has the permission in it?

**Comment by noel-yeldos** [comment]
@lache-melvin working on #5122 and I'm able to reliably recreate this issue. Syncing causes the permission change in 4D to be applied but refreshing page seems to revert user to previous permission. E.g user has permission -> change user to not have permission in 4D -> go to app and sync -> user does not have permission when performing action -> refresh page and the user has permission again

**Comment by aimee-mcneil-melville** [comment]
I can reliable recreate this, changing stores and logging out/in prevented recreation of the bug.
Have tested and it is fixed with #6437

**Comment by Chris-Petty** [comment]
@aimee-mcneil-melville We'd normally wait for the PR to be merged first before closing the issue! But seeing as it's approved will just leave it closed heh

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.6.1-RC4` - just a note, i took these testing steps from the PR but they seem to be mostly about another issue being fixed. Will test with cold chain equipment area just like in the original issue

This is the method I found to reliably recreate the bugs:

- [ ] Take note of your store, and log out of OMS
- [ ] Check mSupply permissions for this store - ensure 'Server Admin' is on (easy example to test 6006)
- [ ] Log into OMS - it should default to the same store
- [ ] Go to settings, you should have a 5+ settings available
- [ ] in mSupply, turn OFF the Server Admin permission - _do not change store or log out & back in after this, as these prevented the bug from occurring
- [ ] In OMS, sync. see your settings reduce to just 'Display'
- [ ] Change language -> see only the Display setting remain
